indykite/common_schema.go

```diff
@@ -187,3 +190,36 @@ func deletionProtectionSchema() *schema.Schema {
 		Description: `Whether or not to allow Terraform to destroy the instance. Unless this field is set to false in Terraform state, a terraform destroy or terraform apply that would delete the instance will fail.`,
 	}
 }
+
+func validateProviderOneOf(providerTypes []string) schema.CustomizeDiffFunc {
+	return func(ctx context.Context, d *schema.ResourceDiff, _ any) error {
+		_ = ctx
+
+		providers := d.Get(providersKey).([]any)
+		for i, p := range providers {
+			if p == nil {
+				continue
+			}
+			providerMap := p.(map[string]any)
+			count := 0
+			for _, value := range providerTypes {
+				if _, ok := providerMap[value]; ok && providerMap[value] != nil {
+					switch v := providerMap[value].(type) {
+					case []any:
+						if len(v) > 0 {
+							count++
+						}
+					case []string:
+						if len(v) > 0 {
+							count++
+						}
+					}
+				}
+			}
+			if count != 1 {
+				return fmt.Errorf("exactly one of providers must be specified in providers[%d]", i)
+			}
+		}
+		return nil
+	}
+}
```

Comment:
Style: This is related to provider, it is reading `providersKey` but is specified in `common_schema` file. Why this is not in `indykite/resource_event_sink.go` where it is used and only there it will be used?